velox/expression/Expr.cpp

```diff
@@ -1239,6 +1239,18 @@ bool Expr::applyFunctionWithPeeling(
     newRows = translateToInnerRows(applyRows, *decoded, newRowsHolder);
     context.saveAndReset(saver, rows);
     setDictionaryWrapping(*decoded, rows, *firstWrapper, context);
+
+    // 'newRows' comes from the set of row numbers in the base vector. These
```

Comment:
do we also need to add similar logic in Expr::peelEncodings()

Reply:
Expr::peelEncodings() already has similar logic:

```
  int numPeeled = 0;
  for (int i = 0; i < peeledVectors.size(); ++i) {
    auto& values = peeledVectors[i];
    if (!values) {
      continue;
    }
    if (!constantFields.empty() && constantFields[i]) {
      context.setPeeled(
          i, BaseVector::wrapInConstant(rows.size(), rows.begin(), values));
    } else {
      context.setPeeled(i, values);
      ++numPeeled;
    }
  }
```

Reply:
@mbasmanova took another look at this, seems like rows.size() is used (in the above code block) for the constant wrap instead of newRows.size(), which can be greater than rows.size(). Hence it can still run into the same bug however, when FieldReference is used to extract the the field, it re-sizes the vector to the correct size so we never see this bug manifest.

Reply:
@bikramSingh91 Thank you for taking a closer look. Should we still fix this bug? Would extracting peeling logic into a separate component help?

CC: @kagamiori

Reply:
@mbasmanova Yes, I plan to consolidate the peeling logic for peelEncoding, applyFunctionWithPeeling and Cast Expr's peeling logic into one single class which would do away with these inconsistencies. Will share more concrete details soon.

Reply:
Thank would great. Looking forward to this refactoring. I hope this component will be useful in map_from_entries Presto function as well. CC: @darrenfu